Src/Plugins/Security/SEC101_003.GoogleApiKeyValidator.cs

```diff
@@ -61,8 +61,10 @@ public static string IsValidDynamic(ref string fingerprint, ref string message)
             const string Deleted = "RequestDenied: This API project was not found. This API project may have been deleted or may not be authorized to use this API. You may need to enable the API under APIs in the console).";
             const string KeyNotAuthorized = "RequestDenied: This API key is not authorized to use this service or API.";
             const string ProjectNotAuthorized = "RequestDenied: This API project is not authorized to use this API.";
+            const string IpNotAuthorized = "RequestDenied: This IP, site or mobile application is not authorized to use this API key.";
```

Comment:
>IpNotAuthorized [](start = 25, length = 15)

This is not the right change. 'NoMatch' indicates a false positive. Both these conditions indicate that we *did* in fact find a match and that the key was actually secured in some way. We can't take this change, we can discuss more offline.

Reply:
I could add Pass to the ValidationState and in the SearchSkimmer, if we get a pass, we change the kind to pass as well. Should we leave error as warning or should we change as well?

---
In reply to: [578522731](https://github.com/microsoft/sarif-pattern-matcher/pull/229#discussion_r578522731) [](ancestors = 578522731)